docs/howto/tracing.md

```diff
@@ -138,6 +174,101 @@ Note that the `charm_tracing` charm library auto-instruments all public function
 of the decorated charm class. `ops[tracing]` doesn't do that, and you are expected
 to create custom spans and events using the OpenTelemetry API where that makes sense.
 
+<!--
+## Trace a charm library
+
+> Hints:
+OTEL API usage.
+A (dummy) charm to test the charm lib.
+
+- At the top of your charm library, `import opentelemetry.trace`.
+- After the imports in your charm library, create the tracer object as
+  `tracer = opentelemetry.trace.get_tracer(name)` where the name could be your
+  charm library name, or Python module `__name__`.
+- See the [Custom spans and events](custom-spans-and-events) section above to
+  create OpenTelemetry spans and events in the key places in your charm library.
+-->
+
+<!--
+## Include tracing in a new charm
+-->
+
+<!--
+## View the trace data
+
+> Hints:
+How to install COS (link), relate your app to COS, where to point browser,
+screenshot of a trace in Grafana, some pointer about what's on the screenshot
+and how manual instrumentation (above) connects to that.
+-->
+
+## Test the feature
+
+A note that ops and charm libraries are already covered, maybe there's
+nothing left for the charm to do.
+
+Relation setup is already validated by the fact that your charm doesn't crash in other tests.
+
+If you've added custom instrumentation, it is because something important is recorded.
+It's easier to cover this with unit tests.
+
+### Write unit tests
+
+The {py:attr}`trace_data <ops.testing.Context.trace_data>` attribute
+of the {py:class}`ops.testing.Context` class of the testing framework 
+contains the list of finished spans.
```

Comment:
We generally write this like:

```suggestion
The [](ops.testing.Context.trace_data) attribute
of the [](ops.testing.Context) class of the testing framework 
contains the list of finished spans.
```

If `{py:}` is better for reason, I'm not opposed to switching to that, but I think it is nice if we're reasonably consistent across our docs.

Reply:
move to short syntax.